lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
Also, there is another changes that was not provide, to add default flags to each compiler defined in a compiler class.  For example, I want to see smth like this:
- compiler:
    environment: {}
    extra_rpaths: []
    flags: {}
    modules: []
    operating_system: rhel7
    paths:
      cc: /usr/bin/xlc
      cxx: /usr/bin/xlc++
      f77: /usr/bin/xlf
      fc: /usr/bin/xlf2003
      fflags: -qzerosize -qfixed
    spec: xl@13.1
    target: ppc64le
- compiler:
    environment: {}
    extra_rpaths: []
    flags: {}
    modules: []
    operating_system: rhel7
    paths:
      cc: /usr/bin/xlc_r
      cxx: /usr/bin/xlc++_r
      f77: /usr/bin/xlf_r
      fc: /usr/bin/xlf2003_r
      fflags: -qzerosize -qfixed
    spec: xl_r@13.1
    target: ppc64le

Pls notice the fflags, which I want to be used by cmake, for example, when it detects the environment for a specific package and before it generates the makefiles to build that package.  The reason for the fflags is that cmake uses Fortran test files that do not always conform to the Fortran spec and the XL Fortran compiler fails to build them w/out those flags.  I have the code changes for that in lib/spack/spack/compilers/__init__.py.

IMO, the best thing is to push for these updates in small steps, i.e. first update the infrastructure such that we have what we need, and then start pushing package.py file changes, one at a time.  Of rouse, based on the latest version of spack.